src/Dialogs/MapItemListDialog.cpp

```diff
@@ -345,20 +351,27 @@ ShowMapItemListDialog(const MapItemList &list,
   switch (list.size()) {
   case 0:
     /* no map items in the list */
-    return;
+    return false;
 
   case 1:
     /* only one map item, show it */
     ShowMapItemDialog(*list[0], airspace_warnings);
-    break;
+    return false;
 
   default:
     /* more than one map item: show a list */
 
     int i = ShowMapItemListDialog(list, dialog_look, look,
                                   traffic_look, final_glide_look, settings);
-    assert(i >= -1 && i < (int)list.size());
+
+    assert(i >= -2 && i < (int)list.size());
+
     if (i >= 0)
       ShowMapItemDialog(*list[i], airspace_warnings);
+
+    /* setting of map item list was done */
+    if(i == -2) return true;
```

Comment:
What is this `-2`?

Reply:
I'm thinking about setting up some constants for it. The return is usually an index of the mapitemlist and -1 if the index does not exist, -2 is now the return for ShowMapItemDialog if a new settings for the list has been made.